arbor/morph/embed_pwlin.cpp

```diff
@@ -304,11 +355,52 @@ embed_pwlin::embed_pwlin(const arb::morphology& m) {
             data_->area[bid].push_back(p0, p1, rat_element<2, 0>(area_0, area_half, area_1));
             area_0 = area_1;
 
-            // (Test for positive dx explicitly in case r0 is zero.)
-            double ixa_half = ixa_0 + (dx>0? dx/(pi*r0*(r0+r1)): 0);
-            double ixa_1 = ixa_0 + (dx>0? dx/(pi*r0*r1): 0);
-            data_->ixa[bid].push_back(p0, p1, rat_element<1, 1>(ixa_0, ixa_half, ixa_1));
-            ixa_0 = ixa_1;
+            if (dx>0) {
+                double ixa_0 = 0;
+                double ixa_half = dx/(pi*r0*(r0+r1));
+                double ixa_1 = dx/(pi*r0*r1);
+
+                if (r0==0 && r1==0) {
+                    // ixa is just not defined on this segment; represent with all
+                    // infinite node values.
+                    ixa_0 = INFINITY;
+                    ixa_half = INFINITY;
+                    ixa_1 = INFINITY;
+                }
+                else if (r0==0) {
+                    ixa_0 = -INFINITY;
+                    ixa_half = -dx/(pi*r1*r1);
+                    ixa_1 = 0;
+                }
+
```

Comment:
This is the case covered in lines 371-373; with r0==0, we pick the interpolating values so that we end up at zero at x==1, from a negative infinite start at x==0 (any finite offset added to this would also work, but this is the simplest).

Reply:
I think I meant `(r1==0 && r0!=0)`, but I guess the default values of `ixa_0`, `ixa_half` and `ixa_1` hold for that case.

Reply:
Yes — the code presumes IEEE semantics, which should give us `ixa_0==0`, `ixa_half==dx/(pi*r0*r0)`, and `ixa_1==INFINITY`.